alvr/client_core/cpp/srgb_correction_pass.cpp

```diff
@@ -0,0 +1,51 @@
+#include "srgb_correction_pass.h"
+#include "utils.h"
+#include <cmath>
+#include <cstdint>
+#include <memory>
+#include <sys/types.h>
+
+using namespace std;
+using namespace gl_render_utils;
+
+namespace {
+const string SRGB_CORRECTION_FRAGMENT_SHADER = R"glsl(#version 300 es
+        #extension GL_OES_EGL_image_external_essl3 : enable
+        precision highp float;
+
+        uniform samplerExternalOES tex0;
+        in vec2 uv;
+        out vec4 color;
+
+        const float DIV12 = 1. / 12.92;
+        const float DIV1 = 1. / 1.055;
+        const float THRESHOLD = 0.04045;
+        const vec3 GAMMA = vec3(2.4);
+
+        void main()
+        {
+            color = texture(tex0, uv);
+
+            vec3 condition = vec3(color.r < THRESHOLD, color.g < THRESHOLD, color.b < THRESHOLD);
+            vec3 lowValues = color.rgb * DIV12;
+            vec3 highValues = pow((color.rgb + 0.055) * DIV1, GAMMA);
+            color.rgb = condition * lowValues + (1.0 - condition) * highValues;
```

Comment:
Just a question, is `(1.0 - condition)` and just `condition` any different? It's basically either `1.0 - 0 or 1`, so why using subtraction here?

Reply:
it's a selector. If one condition evaluates to 0, the other is 1 and vice versa. I always use this trick to avoid using branches which are very bad for performance in shaders.